presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -186,15 +196,22 @@ public void close()
             return;
         }
         closed = true;
-        probe = null;
-        pageBuilder.reset();
+        reachedProbeEnd = true;
```

Comment:
You definitely want to `probe = null;` in `close()` to let memory go. 

(even if you have `reachedProbeEnd` boolean, you can leave it in whatever state here)